### PR TITLE
Button rabbit

### PR DESCRIPTION
**Pull request by cawasser**
built a UI and a backend that can send plain edn messages (using json encoding over RabbitMQ) and Person and Message messages over Protocol Buffers

### CODE REVIEW

**Review by gpierce3 — COMMENTED** [review]
For configuring RabbitMQ, do we use the /main vhost or a new one?

**Comment by cawasser** [comment]
/main

I’ll add that

C

File: README.md [hunk 1]
```diff
@@ -1,21 +1,98 @@
...
+
+### RabbitMQ Configuration
+
+Prior to running the app, you will need to install (not discussed here) _and_ run RabbitMQ. Then, using  
+the management console at
+
+    localhost:15672
+
+you need to set up the following exchanges:
+
```
Comment (gpierce3) [inline README.md:37]: Mention the vhost too.

File: env/dev/clj/dev.clj [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+(defn rollback
+  "Rollback latest database migration."
+  []
+  (migrations/migrate ["rollback"] (select-keys env [:database-url])))
+
+(defn create-migration
+  "Create a new up and down migration file with a generated timestamp and `name`."
+  [name]
+  (migrations/create name (select-keys env [:database-url])))
```
Comment (gpierce3) [inline env/dev/clj/dev.clj:63]: So a group of debugging aides all in one namespace?  Cool.
Does this only get compiled in a dev build?

**Review by gpierce3 — APPROVED** [review]

File: env/dev/clj/dev.clj (hunk 2 above)
Comment (cawasser) [reply env/dev/clj/dev.clj:63]: That is my understanding

File: README.md (hunk 1 above)
Comment (cawasser) [reply README.md:37]: added in f0b8dc

**Review by bfhub — COMMENTED** [review]
followed https://github.com/cawasser/chocolate/tree/button-rabbit, and verified app functionality. The only thing I could not find is the message traffic in RabbitMQ when clicking the message button.

**Review by austinpaine — APPROVED** [review]
Was able to pull and test and replicate results just fine.  Looks good!

**Pull request merged**